### ISSUE TITLE
make test fails

### ISSUE THREAD

**Issue by ambitslix** [noxxi/p5-io-socket-ssl#129]
Version: 2.081

Is this a reason for concern or should this issue be ignored?

```
make test

PERL_DL_NONLAZY=1 "/usr/perl5/5.32/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t t/external/*.t

t/01loadmodule.t .................. 1/3 # openssl version compiled=0x30000050 linked=0x30000070 -- OpenSSL 3.0.7 1 Nov 2022

# Net::SSLeay version=1.92

# parent IO::Socket::IP version=0.39

t/01loadmodule.t .................. ok   

t/acceptSSL-timeout.t ............. ok     

t/alpn.t .......................... ok   

t/auto_verify_hostname.t .......... ok     

t/cert_formats.t .................. 3/6 # start_SSL failed: Failed to load certificate from file t/certs/server.p12: [format:PEM] error:0480006C:PEM routines::no start line error:0A080009:SSL routines::PEM lib ** [format:DER] error:068000A8:asn1 encoding routines::wrong tag error:0688010A:asn1 encoding routines::nested asn1 error error:0688010A:asn1 encoding routines::nested asn1 error error:0A08000D:SSL routines::ASN1 lib ** [format:PKCS12] error:0308010C:digital envelope routines::unsupported error:1180006B:PKCS12 routines::key gen error error:1180006D:PKCS12 routines::mac generation error error:0308010C:digital envelope routines::unsupported error:1180006B:PKCS12 routines::key gen error error:1180006D:PKCS12 routines::mac generation error error:11800071:PKCS12 routines::mac verify failure **

 

#   Failed test '[PKCS12] ssl connect failed: SSL connect attempt failed error:0A000126:SSL routines::unexpected eof while reading'

#   at t/cert_formats.t line 67.

# start_SSL failed: Failed to load certificate from file t/certs/server_enc.p12: [format:PEM] error:0480006C:PEM routines::no start line error:0A080009:SSL routines::PEM lib ** [format:DER] error:068000A8:asn1 encoding routines::wrong tag error:0688010A:asn1 encoding routines::nested asn1 error error:0688010A:asn1 encoding routines::nested asn1 error error:0A08000D:SSL routines::ASN1 lib ** [format:PKCS12] error:0308010C:digital envelope routines::unsupported error:1180006B:PKCS12 routines::key gen error error:1180006D:PKCS12 routines::mac generation error error:0308010C:digital envelope routines::unsupported error:1180006B:PKCS12 routines::key gen error error:1180006D:PKCS12 routines::mac generation error error:11800071:PKCS12 routines::mac verify failure error:0308010C:digital envelope routines::unsupported error:1180006B:PKCS12 routines::key gen error error:1180006D:PKCS12 routines::mac generation error error:11800071:PKCS12 routines::mac verify failure **

 

#   Failed test '[PKCS12_enc] ssl connect failed: SSL connect attempt failed error:0A000126:SSL routines::unexpected eof while reading'

#   at t/cert_formats.t line 67.

# Looks like you failed 2 tests of 6.
```

Thanks much!

**Comment by noxxi** [comment]
I cannot reproduce the problem with Perl 5.34 and OpenSSL 3.0.2. 
Can you check with your OpenSSL version that you are able to load the test certificate with openssl pkcs12, i.e.

```
$ openssl pkcs12 -in t/certs/server.p12 -passin pass:"" -passout pass:"foo"
$ openssl pkcs12 -in t/certs/server_enc.p12 -passin pass:"bluebell" -passout pass:"foo"
```

Have you any other OpenSSL version you can use to check if the test succeeds?

**Comment by twata1** [comment]
I vaguely remember, but perhaps this may be resolved in Net::SSLeay 1.93_01 and above.
